graph/patterns/change-tracking.md

```diff
@@ -0,0 +1,163 @@
+    - `$filter` to influence the scope of changes returned.
+    - `$expand` to include linked resources with the set of changes.
+    - `$top` parameter to influence the size of the set of change records.
+  
+    These query parameters MUST be encoded into subsequent `@odata.nextLink` or `@odata.deltaLink`, such that the same options are preserved through the call sequence without callers respecifying them, which MUST NOT be allowed. OData query parameters must be honored in full, or a 400-error returned.
+- Making a sequence of calls to a delta function followed by the opaque URLs in the `nextLink` and `deltaLink` MUST guarantee that the data at the start time of the call sequence and all changes to the data thereafter will be returned at least once. It is not necessary to avoid duplicates in the sequence. When the delta function is returning changes, they MUST be sequenced chronologically refer to [public documentation](https://learn.microsoft.com/en-us/graph/delta-query-overview?view=graph-rest-1.0) for more details.
+- The delta function can be bound to a collection, as with
+`/users/delta` that returns the changes to the users' collection
+  or to some logical parent resource, where the change records are implied to be relative to all collections contained within the parent, for example
+  `/me/planner/all/delta` – this returns changes to any resource within planner that a user is subscribed to as a heterogenous collection.
+- API producers might use `$skipToken` and `$deltaToken` within their implementations of `nextLink` and `deltaLink`, however the URLs are defined as being opaque and the existence of the tokens MUST NOT be documented.   It is not a breaking change to modify the structure of `nextLinks` or `deltaLinks`.- 
+- `nextLink` and `deltaLink` URLs are valid for a specific period before the client application needs to run a full synchronization again.For `nextLink`, a minimal validity time should be 1 hour. For `deltaLink`, a minimal validity time should be seven days. When a link is no longer valid it must return a standard error with a 410 GONE response code.
+- Although this capability is similar to the OData `$delta` feed capability, it is a different construct. Microsoft Graph APIs MUST provide change tracking through the delta function and MUST NOT implement the OData `$delta` feed when providing change tracking capabilities to ensure the uniformity of the API experience.
+- The Graph delta payload format has some deviations from the OData 4.01 change tracking format to simplify parsing, for example the context annotation is removed.
+  
+
+## Alternatives
+
+- Change notifications pattern with rich payloads – for use cases where API consumers would find calling back into Microsoft Graph onerous and absolute integrity guarantees are less critical.
+
+
+## Examples
+
+### Change tracking on entity set
+
+```
+<Function Name="delta" IsBound="true" ags:OwnerService="Microsoft.DirectoryServices">
```

Comment:
Should these be the **public** CSDLs rather than workload CSDLs?
If so remove any ags annotations, and coerce to the microsoft.graph namespace for type names.

Reply:
updated